### PR TITLE
Implement "Independent C-button Controls" option

### PR DESCRIPTION
**Pull request by Bezier89**
based on the equivalently named option in [Parallel N64](https://github.com/libretro/parallel-n64)

Resolves https://github.com/libretro/mupen64plus-libretro-nx/issues/22

Edit: this PR leaves the option disabled by default, so should be no change in behavior unless explicitly set

### CODE REVIEW

**Comment by mudlord** [comment]
Excellent, I'm so glad you ported it over. Ironically, I was going to port it over today, lol, because there is some games I want to get into playing ASAP.

File: libretro/libretro.c [hunk 1]
```diff
@@ -856,6 +851,13 @@ void update_variables()
             u_cbutton = RETRO_DEVICE_ID_JOYPAD_X;
     }
 
+    var.key = CORE_NAME "-alt-map";
+    var.value = NULL;
+    if (environ_cb(RETRO_ENVIRONMENT_GET_VARIABLE, &var) && var.value)
+    {
+        alternate_mapping = strcmp(var.value, "False");
```
Comment (m4xw) [inline libretro/libretro.c]: alternate_mapping = !strcmp(var.value, "False") ? 0 : 1;

Comment (Bezier89) [reply libretro/libretro.c]: Sure. What I had works, but I can also appreciate wanting consistency with the rest of the file.

**Comment by m4xw** [comment]
I consider this for merge in the cycle sometime next week.
Will need some other small changes, also the prescale defines are relevant for the LLE branch

**Comment by Bezier89** [comment]
@m4xw Got it. Went ahead and added those back. Does that mean this core is getting LLE at some point? :)

**Comment by m4xw** [comment]
https://github.com/libretro/mupen64plus-libretro-nx/tree/lle

**Comment by m4xw** [comment]
Just a reminder I didn't forget about it, but was kinda busy the last days.
Hope I find time this week.

**Comment by mudlord** [comment]
Cool, this gives me time to look over the LLE branch you did.

**Pull request merged**